avm/res/hybrid-container-service/provisioned-cluster-instance/main.bicep

```diff
@@ -0,0 +1,361 @@
+@description('Optional. AAD profile for the connected cluster.')
+param aadProfile aadProfileType?
+
+import { arcAgentProfileType } from 'br/public:avm/res/kubernetes/connected-cluster:0.1.1'
+@description('Optional. Arc agentry configuration for the provisioned cluster.')
+param arcAgentProfile arcAgentProfileType = {
+  agentAutoUpgrade: 'Enabled'
+}
+
+import { oidcIssuerProfileType } from 'br/public:avm/res/kubernetes/connected-cluster:0.1.1'
+@description('Optional. Open ID Connect (OIDC) Issuer Profile for the connected cluster.')
+param oidcIssuerProfile oidcIssuerProfileType = { enabled: false }
+
+import { securityProfileType } from 'br/public:avm/res/kubernetes/connected-cluster:0.1.1'
+@description('Optional. Security profile for the connected cluster.')
+param securityProfile securityProfileType = {
+  workloadIdentity: {
+    enabled: false
+  }
+}
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.res.hybcontsvc-provclustinst.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource kv 'Microsoft.KeyVault/vaults@2023-07-01' existing = if (empty(linuxProfile) && !empty(keyVaultName)) {
+  name: keyVaultName!
+}
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
+  name: 'temp-${name}'
+  location: location
+  tags: tags
+}
+
+resource generateSSHKey 'Microsoft.Resources/deploymentScripts@2020-10-01' = if (empty(linuxProfile)) {
+  name: 'generateSSHKey-${name}'
+  location: location
+  tags: tags
+  kind: 'AzurePowerShell'
+  identity: {
+    type: 'UserAssigned'
+    userAssignedIdentities: {
+      '${managedIdentity.id}': {}
+    }
+  }
+  properties: {
+    azPowerShellVersion: '8.0'
+    retentionInterval: 'P1D'
+    scriptContent: '''
+      # Create temp directory in a known location
+      $tempDir = "/tmp/sshkeys"
+      New-Item -ItemType Directory -Path $tempDir -Force
+      Set-Location $tempDir
+      # Generate SSH key pair using ssh-keygen
+      ssh-keygen -t rsa -b 4096 -f ./key -N '""' -q
+      # Read the generated keys
+      $publicKey = Get-Content -Path "./key.pub" -Raw
+      $privateKey = Get-Content -Path "./key" -Raw
+      # Clean up temp files
+      Remove-Item -Path "./key*" -Force
+      Remove-Item -Path $tempDir -Force -Recurse
+      # Set output
+      $DeploymentScriptOutputs = @{}
+      $DeploymentScriptOutputs['publicKey'] = $publicKey
+      $DeploymentScriptOutputs['privateKey'] = $privateKey
+    '''
+  }
```

Comment:
I would highly recommend moving this content a dedicated script in a `scripts` subfolder (e.g. `scripts/generateKey.sh`) and load it via `loadTextContent()` as seen here

https://github.com/Azure/bicep-registry-modules/blob/3385341b7cdd437703e65ace083ac5b86be990ce/avm/ptn/dev-ops/cicd-agents-and-runners/main.bicep#L803

It does not change how your module works - but - makes it possible to run & potentially troubleshoot the script's content without needing to rely on the deployment of the deploymentScript.